pavelib/stanford_i18n.py

```diff
@@ -0,0 +1,282 @@
+    print make_djangojs_cmd
+    sp.check_call(make_djangojs_cmd, shell=True)
+
+    djangojs_file = CONFIG.source_messages_dir.joinpath('stanford_djangojs.po')
+    os.rename(CONFIG.source_messages_dir / 'djangojs.po', djangojs_file)
+    files_to_clean.add(djangojs_file)
+
+    for filename in files_to_clean:
+         pofile = polib.pofile(filename)
+         fix_header(pofile)
+         fix_metadata(pofile)
+         pofile.save()
+
+    os.rename(CONFIG.source_messages_dir / 'django-saved.po', CONFIG.source_messages_dir / 'django.po')
+    os.rename(CONFIG.source_messages_dir / 'djangojs-saved.po', CONFIG.source_messages_dir / 'djangojs.po')
+
+@task
+@needs(
+    "pavelib.prereqs.install_prereqs",
+)
+def stanfordi18n_extract_theme(options):
+    """
+    Extract strings that need translation.
+    """
+    files_to_clean = set()
+    babel_mako_cfg = '/edx/app/edxapp/themes/comprehensive/conf/locale/babel_mako.cfg'
+    template_dir = 'lagunita/lms/templates/'
+    theme_file = CONFIG.source_messages_dir.joinpath('theme.po')
+    babel_cmd_template = (
+        'pybabel extract --mapping={config} '
+        '--add-comments="Translators:" --keyword="interpolate" '
+        '--output={output} {template_dir}'
+    )
+    babel_mako_cmd = babel_cmd_template.format(
+        config=babel_mako_cfg,
+        output='../../edx-platform' / theme_file,
+        template_dir=template_dir,
+    )
+    print babel_mako_cmd
+    sp.check_call(babel_mako_cmd, cwd='/edx/app/edxapp/themes/comprehensive', shell=True)
+    files_to_clean.add(theme_file)
+
+    # Monkeypatch upstream function
+    segment.segment_pofile = segment_pofile_lazy
+
+    # Segment out tos strings into separate files
+    segmented_tos = segment.segment_pofiles(CONFIG, CONFIG.source_locale)
+    files_to_clean.update(segmented_tos)
+
+    for filename in files_to_clean:
+        pofile = polib.pofile(filename)
+        fix_header(pofile)
+        fix_metadata(pofile)
+        pofile.save()
+
+    print "DONE EXTRACTING THEME"
+
+
+@task
+@needs("pavelib.i18n.i18n_validate_transifex_config")
+def stanfordi18n_transifex_push():
+    """
+    Push source strings to Transifex for translation
+    """
+    sp.check_call('tx push -s -r stanford-openedx.theme', shell=True)
+    sp.check_call('tx push -s -r stanford-openedx.tos', shell=True)
+    sp.check_call('tx push -s -r stanford-openedx.privacy', shell=True)
+    sp.check_call('tx push -s -r stanford-openedx.honor', shell=True)
+    sp.check_call('tx push -s -r stanford-openedx.copyright', shell=True)
+    sp.check_call('tx push -s -r stanford-openedx.mako', shell=True)
+    sp.check_call('tx push -s -r stanford-openedx.django', shell=True)
+    sp.check_call('tx push -s -r stanford-openedx.djangojs', shell=True)
+    sp.check_call('tx push -s -r stanford-openedx.underscore', shell=True)
+    print "PUSHED SOURCE FILES TO TRANSIFEX"
+
+    # merge in existing platform translations
+    for lang in CONFIG.translated_locales:
```

Comment:
Can the body of this loop be refactored to make use of a helper functions?

It looks like they all follow a similar form:
- get template string
- get filenames
- format file/source/output into the command name
- print the command
- run `check_call`

As is, this function is already pretty massive and I'm having a bit of trouble tracing through it, but I suspect we're doing the same thing over and over and this would benefit from abstracting the logic.